addons/website_sale_wallet/models/partner.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from odoo import fields, models, api
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    wallet_ids = fields.One2many("wallet", "partner_id")
+
+    current_website = fields.Many2one('website', compute='_compute_current_website', store=False)
+    current_wallet_id = fields.Many2one("wallet", compute="_compute_default_wallet_id", store=False)
+
+    def _compute_current_website(self):
+        for record in self:
+            record.current_website = record.env['website'].get_current_website()
+
+    @api.depends('current_website')
+    def _compute_default_wallet_id(self):
+        for record in self:
+            record.current_wallet_id = self.wallet_ids.search(
+                ['&', ('website_id', '=', self.current_website.id), ('partner_id', '=', self.id)], limit=1
+            )
+
+    def add_gift_card_to_current_website(self, gift_card_id):
```

Comment:
a public function that anybody can call to add any card_id ? without check of code or ... ?
wrong design

Reply:
we check if the card is new or not specific to the website (inside the method).
`python
      if gift_card_id.is_gift_card_new_for(self.current_website):
`